src/routes/friend.js

```diff
@@ -1,88 +1,197 @@
+ */
+async function addToFriendLists(student1, student2) {
+  await addToFriendList(student1, student2);
+  await addToFriendList(student2, student1);
+}
+
+/**
+ * Helper function to add a student to another one friend list.
+ */
+async function addToFriendList(email1, email2) {
+  const student = await Student.findOne({email: email1});
+  student.friends.push(email2);
+  student.save();
+  // 2DO : need to add middleware to throw custom errors.
+}
+
+/**
+ * @author: Jasmin Guay
+ * Endpoint to display all incoming requests.
+ * @param email (string) : email of the student to fetch incoming requests
+ * @return friendRequests (friendRequest[]) : list of friend requests sent to the specified student.
+ */
+router.route("/incoming-requests/:email").get( async (req, res) => {
+  const email = req.params.email.toString();
+
+  FriendRequest.find({
+    receiverEmail: email
+  }).then((requests) => {
+    res.json(requests).status(200);
+  }).catch((err) => {
+    res.json(`Cannot fetch incoming requests for [${email}] - ${err}`);
+  })
+})
+
+/**
+ * @author: Jasmin Guay
+ * Endpoint to display all outgoing requests.
+ * @param email (string) : email of the student to fetch outgoing requests
+ * @return friendRequests (friendRequest[]) : list of friend requests received by the specified student.
+ */
+router.route("/outgoing-requests/:email").get( async (req, res) => {
+  const email = req.params.email.toString();
+
+  FriendRequest.find({
+    senderEmail: email
+  }).then((requests) => {
+    res.json(requests).status(200);
+  }).catch((err) => {
+    res.json(`Cannot fetch outgoing requests for [${email}] - ${err}`);
+  })
+})
+
+/**
+ * @author: Jasmin Guay
+ * Endpoint to delete an outgoing request
+ * @param requestId (string) : id of the request to be deleted
+ * @return friendRequests (friendRequest[]) : list of friend requests received by the specified student.
+ */
+router.route("/cancel-request").post( async (req, res) => {
+    const requestId = req.body.requestId.toString();
+
+  FriendRequest.findOneAndDelete({
+    _id: requestId
+  }).then((requests) => {
+    res.json(requests).status(200);
+  }).catch((err) => {
+    res.json(`Cannot delete request with id [${requestId}] - ${err}`);
+  })
+})
+
+/**
+ * @author: Jasmin Guay
+ * Endpoint to search a user
+ * @param {string} searchInput : text input to search a user (email or username)
+ * @return {username: string, password: string} : minimal object of the user found (if profile is not private)
+ */
+router.route("/search").post( async (req,res) => {
+  const searchInput = req.body.searchInput.toString();
+
+  let user = await StudentRepository.findOneByEmail(searchInput);
+
+  if (user == null) {
+    user = await StudentRepository.findOneByUsername(searchInput);
+  }
+  if (user == null || user.privateProfile) {
+    res.json(`No user found for searchInput [${searchInput}]`).status(404);
+  } else {
+    res.json({username: user.email,email: user.username}).status(200);
```

Comment:
should be username: user.username and email: user.email
they are reversed right now